src/sentry/integrations/github/client.py

```diff
@@ -118,7 +118,8 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
             elif msg == "Not Found":
                 logger.warning(f"The Github App does not have access to {repo_full_name}.")
             else:
-                logger.exception("An unknown error has ocurred.")
+                # Raise it so we can stop hammering the API
+                raise (e)
```

Comment:
We raise it inside of `get_tree` and `get_cache_repo_files` will also bubble it up. `get_trees_for_org` will handle it.